routers/repo/pull.go

```diff
@@ -673,6 +673,23 @@ func MergePullRequest(ctx *context.Context, form auth.MergePullRequestForm) {
 		return
 	}
 
+	if form.MergeWhenChecksSucceed {
```

Comment:
Maybe checks are already successful at this point.

Reply:
You mean as a possible race condition? Or if the user pusehs, updates the pr site, then does nothing, meanwhile the checks pass and on the pr page it would still say "merge when checks succeed"?
Either way, I'll add a check to prevent these kinds.